### PR TITLE
Improve/redux

### PR DESCRIPTION
**Pull request by daruj**
Summary: Implemented best practices for redux:

- Ducks
- Stateless Components
- Improved structure
- Promise middleware

### CODE REVIEW

File: src/api/fake/user.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { getToken } from '../auth_token';
+
+const UserApiCalls = {
+  getUser: () => {
+    let promise;
+    if (!getToken()) {
```
Comment (ismatim) [inline src/api/fake/user.js]: pondría un `if(getToken)) return new Promise.
Te evitas el nested sin `else` y luego `return new Promise (...rejected...)

File: src/api/fake/user.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { getToken } from '../auth_token';
+
+const UserApiCalls = {
+  getUser: () => {
+    let promise;
```
Comment (ismatim) [inline src/api/fake/user.js]: No necesitas crearla.

File: src/components/SignInPage/index.jsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import SignInPageContainer from './SignInPage.container';
+
+export default SignInPageContainer;
+
```
Comment (ismatim) [inline src/components/SignInPage/index.jsx]: Estos espacios los sacaría.

File: src/redux/modules/reducers.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { combineReducers } from 'redux';
+
+import authentication from './authentication';
+import user from './user';
+
+export default combineReducers({
+  authentication,
+  user
+});
```
Comment (ismatim) [inline src/redux/modules/reducers.js]: A todos los fin de archivo les dejaría una línea.

File: src/api/fake/user.js (hunk 1 above)
Comment (daruj) [inline src/api/fake/user.js]: I wanted to fake a loading time.. and that's why I can't do that. take I look, there is a setTimeOut that does not let resolve this the way you mentioned.

File: src/components/SignInPage/index.jsx (hunk 3 above)
Comment (daruj) [inline src/components/SignInPage/index.jsx]: That's an eslint rule!

File: src/api/fake/user.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    if (!getToken()) {
+      // Get error when the token is not saved in the localStorage
+      promise = new Promise((resolve, reject) => setTimeout(() => reject({
+        error: {
+          message: 'You are not logged in'
+        }
+      }), 3000));
```
Comment (ismatim) [inline src/api/fake/user.js]: hardcoded ?

Comment (daruj) [inline src/api/fake/user.js]: yes, its a fake api

**Pull request merged**